extensions/machine-learning-services/src/widgets/configTable.ts

```diff
@@ -0,0 +1,182 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ApiWrapper } from '../common/apiWrapper';
+import { ServerConfigManager } from '../serverConfig/serverConfigManager';
+import * as constants from '../common/constants';
+
+export class ConfigTable {
+	private _statusTable: azdata.DeclarativeTableComponent;
+
+	constructor(private _apiWrapper: ApiWrapper, private _serverConfigManager: ServerConfigManager, private _modelBuilder: azdata.ModelBuilder, private _loadingComponent: azdata.LoadingComponent) {
+		this._statusTable = this._modelBuilder.declarativeTable()
+			.withProperties<azdata.DeclarativeTableProperties>(
+				{
+					columns: [
+						{ // Config
+							displayName: constants.mlsConfigTitle,
+							valueType: azdata.DeclarativeDataType.component,
+							isReadOnly: true,
+							width: 175,
+							headerCssStyles: {
+								'border': 'none',
+								'background-color': '#FFFFFF',
+								...constants.cssStyles.tableHeader
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						},
+						{ // Status icon
+							displayName: constants.mlsConfigStatus,
+							ariaLabel: constants.mlsConfigStatus,
+							valueType: azdata.DeclarativeDataType.component,
+							isReadOnly: true,
+							width: 25,
+							headerCssStyles: {
+								'border': 'none',
+								'background-color': '#FFFFFF'
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						},
+						{ // Action
+							displayName: '',
+							valueType: azdata.DeclarativeDataType.component,
+							isReadOnly: true,
+							width: 150,
+							headerCssStyles: {
+								'border': 'none',
+								'background-color': '#FFFFFF',
+								...constants.cssStyles.tableHeader
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						}
+					],
+					data: [],
+					ariaLabel: constants.mlsConfigTitle
+				})
+			.component();
+	}
+
+	/**
+	 * Returns the config table component
+	 */
+	public get component(): azdata.DeclarativeTableComponent {
+		return this._statusTable;
+	}
+
+	/**
+	 * Refreshes the config table
+	 */
+	public async refresh(): Promise<void> {
+		this._loadingComponent.updateProperties({ loading: true });
+		let connection = await this.getCurrentConnection();
+		const speesConfig = await this.addTableRow(constants.mlsExternalExecuteScriptTitle,
```

Comment:
>speesConfig [](start = 8, length = 11)

What's spees? That doesn't appear to be used anywhere else

Reply:
I'll rename it. sp for external scripts

---
In reply to: [362905485](https://github.com/microsoft/azuredatastudio/pull/8796#discussion_r362905485) [](ancestors = 362905485)